java-api/src/main/java/com/sap/cloud/security/config/FileSystemAccessor.java

```diff
@@ -0,0 +1,39 @@
+package com.sap.cloud.security.config;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Properties;
+
+/**
+ * The interface for K8s File system access.
+ */
+public interface FileSystemAccessor {
+
+
+    /**
+     * Gets all bound Xsuaa service instances
+     *
+     * @return the array of files
+     */
+    @Nullable
+    File[] getXsuaaBindings();
+
+    /**
+     * Gets all bound IAS service instances
+     *
+     * @return the array of files
+     */
+    @Nullable
+    File[] getIasBindings();
+
+    /**
+     * Extracts xsuaa binding properties from a single binding.
+     *
+     * @param bindings the service instance bindings
+     * @return the file array of service configuration properties
+     */
+    @Nullable
+    File[] extractXsuaaBindingFiles(File[] bindings);
```

Comment:
i don't understand this one... i would just remove that, it can be implemented as part of K8S implementation.